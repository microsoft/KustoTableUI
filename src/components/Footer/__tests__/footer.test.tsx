import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { Footer } from '..';

describe('Footer Snapshot UI Tests', () => {
  let component: ReactTestRenderer;

  afterEach(() => component.unmount());

  test.each`
    buildVersion     | mailTo               | title             | scenarioName
    ${null}          | ${null}              | ${null}           | ${'Null BuildVersion, MailTo and Title'}
    ${undefined}     | ${undefined}         | ${undefined}      | ${'Undefined BuildVersion, MailTo and Title'}
    ${''}            | ${''}                | ${''}             | ${'Empty String BuildVersion, MailTo and Title'}
    ${'123.123.134'} | ${'email@email.com'} | ${'KustoTableUI'} | ${'Normal BuildVersion, MailTo and Title'}
  `(
    'Test Footer Rendering $scenarioName',
    ({ buildVersion, mailTo, title }) => {
      component = renderer.create(
        <Footer buildVersion={buildVersion} mailTo={mailTo} title={title} />,
      );
      const footer = component.toJSON();
      expect(footer).toMatchSnapshot();
    },
  );
});
