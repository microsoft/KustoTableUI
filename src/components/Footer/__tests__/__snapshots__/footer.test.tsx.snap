// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot UI Tests Test Footer Rendering Empty String BuildVersion, MailTo and Title 1`] = `
<div
  className="footer-container"
>
  <div
    className="footer-container-left"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
     2023 - 
     [
    <a
      href="mailto:"
    >
      Support
    </a>
    ]
  </div>
</div>
`;

exports[`Footer Snapshot UI Tests Test Footer Rendering Normal BuildVersion, MailTo and Title 1`] = `
<div
  className="footer-container"
>
  <div
    className="footer-container-left"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
     2023 - 
    KustoTableUI
     [
    <a
      href="mailto:email@email.com"
    >
      Support
    </a>
    ]
  </div>
</div>
`;

exports[`Footer Snapshot UI Tests Test Footer Rendering Null BuildVersion, MailTo and Title 1`] = `
<div
  className="footer-container"
>
  <div
    className="footer-container-left"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
     2023 - 
     [
    <a
      href="mailto:"
    >
      Support
    </a>
    ]
  </div>
</div>
`;

exports[`Footer Snapshot UI Tests Test Footer Rendering Undefined BuildVersion, MailTo and Title 1`] = `
<div
  className="footer-container"
>
  <div
    className="footer-container-left"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
     2023 - 
     [
    <a
      href="mailto:"
    >
      Support
    </a>
    ]
  </div>
</div>
`;
